### ISSUE TITLE
Abstract property implemented in concrete class is missed

### ISSUE THREAD

**Issue by m3at** [microsoft/pyright#706]
**Describe the bug**
When a concrete class instantiate a property, pyright wrongfully indicate `[Pyright (reportGeneralTypeIssues)] [E] Cannot instantiate abstract class`

**Code to reproduce**
```python
from abc import ABC, abstractmethod

class Abstract(ABC):
    @property
    @abstractmethod
    def bar(self):
        pass

    @abstractmethod
    def foo(self) -> None:
        pass

class Concrete(Abstract):

    bar = 42

    def foo(self):
        return

# We get the warning here
a = Concrete()
print(a.bar)
# >>> 42
```

Note that if we define `bar` that way instead, we have no issue:
```python
@property
def bar(self):
    return 42
```

**VS Code extension or command-line**
Latest pyright as of 2020/06/03

**Comment by erictraut** [comment]
Thanks for the issue.

In this case, it's working as intended. Your base class is indicating that subclasses must implement a method that implements a property getter (with no setter or deleter). The subclass you have created does not define such a method. Instead, it defines a class variable of the same name as the abstract method.

I guess it could be argued that the error message in this case is misleading because Python will allow you to instantiate the subclass without generating an exception, but this is still a clear programming error that should be flagged by a type checker.

**Comment by m3at** [comment]
Ah you're right! Thanks for the explanation it make sense, I got a bit confused here.
